src/firefly/js/ui/CompleteButton.jsx

```diff
@@ -33,59 +33,52 @@ function addComponentData(inR={}, registeredComponents={}) {
 }
 
 
-function validUpdate(valid,onSuccess,onFail,registeredComponents, closeOnValid,groupKey,
-                     dialogId, groupsToUse, includeUnmounted= false) {
-    const funcToCall = valid ? onSuccess : onFail;
-    let continueValid= true;
-
-    if (Array.isArray(groupKey)) {
-        const groupsToValidate= groupsToUse();
-        const requestObj = groupsToValidate.reduce( (obj,groupKey) => {
-            obj[groupKey]= getFieldGroupResults(groupKey,includeUnmounted);
-            return obj;
-        },{});
-        if (funcToCall) funcToCall(addComponentData(requestObj,registeredComponents));
-    }
-    else {
-        const request = getFieldGroupResults(groupKey,includeUnmounted);
-        if (funcToCall) continueValid= funcToCall(addComponentData(request,registeredComponents));
-    }
+async function collectInputs(registeredComponents, groupKey, includeUnmounted= false) {
 
-    const stillValid= valid && (continueValid ?? true);
+    if (!groupKey) return [{}, true];   // if no groupKey, then no parameters to collect and therefore pass validation
 
-    if (stillValid && dialogId && closeOnValid) dispatchHideDialog(dialogId);
-}
+    const groups = isFunction(groupKey) ? groupKey() : groupKey;
+    const valid = await validateFieldGroup(groups, includeUnmounted);
 
-function onClick(onSuccess,onFail,registeredComponents, closeOnValid,groupKey,
-                 dialogId,groupsToUse, includeUnmounted, changeMasking, requireAllValid) {
-    if (groupKey) {
-        if (changeMasking) changeMasking(true);
-        validateFieldGroup(groupsToUse(), includeUnmounted)
-            .then( (valid)=> {
-                if (changeMasking) changeMasking(false);
-                validUpdate(valid||!requireAllValid,onSuccess,onFail,registeredComponents,closeOnValid,groupKey,dialogId, groupsToUse, includeUnmounted);
-            });
+    let inputs;
+    if (isFunction(groupKey)) {
+        const groupsToValidate= groupKey();
+        inputs = groupsToValidate.reduce( (obj,gkey) => {
+            obj[gkey]= getFieldGroupResults(gkey,includeUnmounted);
+            return obj;
+        },{});
```

Comment:
I think this should be:
```js
    if (Arrays.isArray(groups)) {
        inputs = groups.reduce( (obj,gkey) => {
            obj[gkey]= getFieldGroupResults(gkey,includeUnmounted);
            return obj;
        },{});
    else {
        inputs = getFieldGroupResults(groupKey,includeUnmounted);
    }
```

Reply:
`ColorDialog.jsx` line 164 has the following:
```jsx
                  <CompleteButton
                        text='Refresh' dialogId='ColorStretchDialog' includeUnmounted={true}
                        groupKey={['colorDialogTabs',RED_PANEL,GREEN_PANEL,BLUE_PANEL]}
                        closeOnValid={false} sx={{pt:.25, pb:1, pl:1}}
                        onSuccess={replot(usedBands)} onFail={invalidMessage}
                    />
```

Reply:
You are right.  I was suppose to go back to change all places that uses an array `groupKey` to a function instead.  But, if you prefer supporting 3 type of groupKey, i'll change it at the completeButton.

Reply:
I actually think we could take out the function. I don’t think we use it. I think it was an idea I had long ago the was never used. 

I hope to get away from the array. Our change in the last couple of years has made array mostly unnecessary (I think). But until we refactor everything  we will not be sure.

Reply:
We used 'func' [here](https://github.com/Caltech-IPAC/firefly/blob/653149882416ce72dada04a861ff7ec612c0ab0d/src/firefly/js/visualize/ui/ImageSearchPanelV2.jsx#L125)
I removed the prop and combine it with `groupKey`.  It makes more sense to me.